js/classes/entities/foreground/surface_snow.js

```diff
@@ -0,0 +1,55 @@
+class SurfaceSnow {
+    constructor() {
+        this.pos = new Vector(0, 0);
+
+        this.animator = new Animator(Precipitation.SPRITESHEET, new Vector(0, 7 * Precipitation.SIZE.x), new Vector(Precipitation.SIZE.x, Precipitation.SIZE.y), 1, 1);
+
+    }
+    static get SPRITESHEET() {
+        return "./sprites/snow_and_rain.png";
+    }
+    static get SIZE() {
+        return new Vector(32, 32);
+    }
+
+    /**
+     * Update the BoundingBox of the Projectile based on the latter's current position.
+     */
+    updateBoundingBox() {
+        this.boundingBox = new BoundingBox(this.pos, Vector.multiply(this.size, Projectile.SCALE));
+    };
+
+    setupBoundingBox() {
+        this.boundingBox = new BoundingBox(this.pos, Vector.multiply(SurfaceSnow.SIZE, 1));
+    }
+    update() {
+        this.velocity.y = PHYSICS.GRAVITY_ACC * GAME.clockTick;
+        setupBoundingBox();
+        GAME.entities.midground.forEach((entity) => {
+            if (this != entity && entity.boundingBox && this.boundingBox.collide(entity.boundingBox)) {
+                this.pos = Vector.blockToWorldSpace(new Vector(this.velocity.y, CHAD.pos.x));
```

Comment:
It looks to me like this class is a work in progress so you may already realize this, but I just wanted to note that the position value that will result from this operation is going to be absolutely useless. I'm not super sure what you were trying to do but I can guarantee that this won't do it.

Reply:
Oh I didn't realize that. Why is that? Also I will remove this class as I found a simpler way of achieving the goal of this class.

Reply:
Sorry, I should have been more specific. To start with, you're passing a Vector that is in world coordinates into the `blockToWorldSpace` method, which expects a block coordinate Vector. Secondly, you're using the y-velocity as the x-coordinate and Chad's x-position as the y-coordinate. It also doesn't make much sense to use velocity at all here since you're assigning this as your snow's position.

Reply:
Oh wow, you're right. That makes no sense. I likely intended to use its changing y position to track snow. I should have waited to push that class, regardless thank you for the explanation. I'll be removing this class in favor of a simpler solution.